docs_espressif/en/additionalfeatures/esp-idf-profiles.rst

```diff
@@ -0,0 +1,71 @@
+.. _esp-idf-profiles:
+
+ESP-IDF VS Code Profiles
+========================
+
+:link_to_translation:`zh_CN:[中文]`
+
+The ESP-IDF extension for VS Code works best with specific companion extensions for full C/C++ language support (like IntelliSense, code navigation, debugging) and CMake integration. To simplify the setup process, we provide official VS Code profiles that bundle recommended extensions and settings.
+
+What is a VS Code Profile?
+--------------------------
+
+A VS Code profile (`.code-profile`) is a collection of extensions, settings, UI state, and keyboard shortcuts. Using an official ESP-IDF profile ensures that you have a curated set of tools optimized for ESP-IDF development without needing to manually install and configure each component.
+
+Available ESP-IDF Profiles
+--------------------------
+
+We offer two primary profiles tailored to different C/C++ language server preferences:
+
+**1. Default Profile (`ESP-IDF`)**
+
+* **C/C++ Tooling**: Uses the **Microsoft C/C++ Extension Pack** (`ms-vscode.cpptools-extension-pack`). This includes the `ms-vscode.cpptools` extension, which provides IntelliSense, debugging, and code Browse features widely used in VS Code for C/C++ development.
+* **Included Key Extensions**:
+    * `espressif.esp-idf-extension` (ESP-IDF)
+    * `ms-vscode.cpptools-extension-pack` (C/C++ Extension Pack)
+    * `ms-vscode.cmake-tools` (CMake Tools)
+* **Recommendation**: This profile is recommended for most users and provides a comprehensive C/C++ development experience based on Microsoft's standard tooling. The internal profile name in VS Code is **`ESP-IDF`**.
+
+**2. Clangd Profile (`ESP-IDF (clangd)`)**
+
+* **C/C++ Tooling**: Uses **`clangd`** (`llvm-vs-code-extensions.vscode-clangd`) as the C/C++ language server. `clangd` is part of the LLVM project and provides features like code completion, diagnostics, and navigation based on the Clang compiler infrastructure.
+* **Included Key Extensions**:
+    * `espressif.esp-idf-extension` (ESP-IDF)
+    * `llvm-vs-code-extensions.vscode-clangd` (clangd)
+    * `ms-vscode.cmake-tools` (CMake Tools)
```

Comment:
I would remove `ms-vscode.cmake-tools` from the clang profile. Since Microsoft does not like their tools to be used for vscode-like IDEs we should separate them from the open source LLVM choice.

Why do we include `ms-vscode.cmake-tools` at all?